File: parse_discovery.js
const fs = require('fs');
const path = require('path');

const demoDir = '/Users/charlotte/Downloads/oria-discovery-demo';
const files = ['eat.html', 'quiet.html', 'two-hours.html', 'local.html', 'walk.html', 'buy.html'];
const data = {};

for (const file of files) {
    const html = fs.readFileSync(path.join(demoDir, file), 'utf8');
    
    const getMatch = (regex) => {
        const match = html.match(regex);
        return match ? match[1].trim() : '';
    };

    const title = getMatch(/<h1>(.*?)<\/h1>/);
    const eyebrow = getMatch(/<div class="eyebrow">(.*?)<\/div>/);
    const intro = getMatch(/<h1>.*?<\/h1>\s*<p>(.*?)<\/p>/s);
    const imgMatch = html.match(/<section class="cover">\s*<img src="(.*?)"/);
    const image = imgMatch ? imgMatch[1].replace('assets/', '/images/discovery/') : '';
    const lede = getMatch(/<p class="lede">(.*?)<\/p>/s);
    
    let quote = null;
    const quoteMatch = html.match(/<blockquote class="quote">(.*?)(?:<small>(.*?)<\/small>)?<\/blockquote>/s);
    if (quoteMatch) {
        quote = {
            text: quoteMatch[1].trim(),
            small: quoteMatch[2] ? quoteMatch[2].trim() : ''
        };
    }

    const sections = [];
    const sectionRegex = /<section class="section">(.*?)<\/section>/gs;
    let secMatch;
    while ((secMatch = sectionRegex.exec(html)) !== null) {
        const secHtml = secMatch[1];
        
        const secLabelMatch = secHtml.match(/<div class="section-label">(.*?)<\/div>/);
        const label = secLabelMatch ? secLabelMatch[1].trim() : '';
        
        const h2Match = secHtml.match(/<h2>(.*?)<\/h2>/);
        const heading = h2Match ? h2Match[1].trim() : '';

        let innerHtml = secHtml.replace(/<div class="section-label">.*?<\/div>\s*<div>/, '');
        innerHtml = innerHtml.replace(/<\/div>\s*$/, '');
        innerHtml = innerHtml.replace(/<h2>.*?<\/h2>/, '');
        
        const cards = [];
        const cardsRegex = /<div class="card">(.*?)<\/div>/gs;
        let cardMatch;
        while ((cardMatch = cardsRegex.exec(innerHtml)) !== null) {
            const cHtml = cardMatch[1];
            const cTitle = (cHtml.match(/<b>(.*?)<\/b>/) || [])[1] || '';
            const cDesc = (cHtml.match(/<p>(.*?)<\/p>/) || [])[1] || '';
            const cMap = (cHtml.match(/<a class="maplink" .*?href="(.*?)".*?>/) || [])[1] || '';
            cards.push({ title: cTitle, description: cDesc, map_url: cMap });
        }
        
        innerHtml = innerHtml.replace(/<div class="cards">.*?<\/div>\s*<\/div>/s, '');
        innerHtml = innerHtml.replace(/<div class="cards">.*?<\/div>\s*$/s, '');
        
        sections.push({ label, heading, bodyHTML: innerHtml.trim(), cards });
    }
    
    const slug = title.replace('.', '').trim();
    data[slug] = { eyebrow, title, image, intro, lede, sections, quote };
}

fs.writeFileSync('/Users/charlotte/Desktop/NGÂN HÀ/CTY TechGalaxy Group/NganHa-WebBooking/src/data/DiscoveryData.json', JSON.stringify(data, null, 2));
console.log('Parsed and saved to DiscoveryData.json');

File: parse_discovery.py
import os
from bs4 import BeautifulSoup
import json

demo_dir = '/Users/charlotte/Downloads/oria-discovery-demo'
files = ['eat.html', 'quiet.html', 'two-hours.html', 'local.html', 'walk.html', 'buy.html']
data = {}

for f in files:
    with open(os.path.join(demo_dir, f), 'r') as file:
        soup = BeautifulSoup(file.read(), 'html.parser')
        
        # Cover
        cover = soup.find(class_='cover')
        img_src = cover.find('img')['src'] if cover.find('img') else ''
        cover_copy = cover.find(class_='cover-copy')
        eyebrow = cover_copy.find(class_='eyebrow').text if cover_copy.find(class_='eyebrow') else ''
        title = cover_copy.find('h1').text if cover_copy.find('h1') else ''
        intro = cover_copy.find('p').text if cover_copy.find('p') else ''
        
        # Article
        article = soup.find(class_='article')
        lede = article.find(class_='lede').text if article.find(class_='lede') else ''
        
        # Quote
        quote_tag = article.find(class_='quote')
        if quote_tag:
            quote_text = quote_tag.contents[0] if isinstance(quote_tag.contents[0], str) else quote_tag.text
            quote_small = quote_tag.find('small').text if quote_tag.find('small') else ''
            quote = {'text': str(quote_text).strip(), 'small': quote_small}
        else:
            quote = None

        # Sections
        sections = []
        for sec in article.find_all(class_='section'):
            label = sec.find(class_='section-label').text if sec.find(class_='section-label') else ''
            content_div = sec.find('div', class_=False) or sec.find_all('div')[-1]
            h2 = content_div.find('h2').text if content_div.find('h2') else ''
            
            # extract all paragraphs and ul/ol that are direct children of content_div but not cards
            body_elements = []
            for child in content_div.children:
                if child.name in ['p', 'ul', 'ol', 'div'] and 'cards' not in child.get('class', []) and child.name != 'h2':
                    body_elements.append(str(child))
            
            # cards
            cards_container = content_div.find(class_='cards')
            cards = []
            if cards_container:
                for card in cards_container.find_all(class_='card'):
                    c_title = card.find('b').text if card.find('b') else ''
                    c_desc = card.find('p').text if card.find('p') else ''
                    map_url = card.find('a', class_='maplink')['href'] if card.find('a', class_='maplink') else ''
                    cards.append({'title': c_title, 'description': c_desc, 'map_url': map_url})
                    
            sections.append({
                'label': label,
                'heading': h2,
                'bodyHTML': ''.join(body_elements).strip(),
                'cards': cards
            })
            
        slug = title.replace('.', '').strip()
        
        data[slug] = {
            'eyebrow': eyebrow,
            'title': title,
            'image': img_src,
            'intro': intro,
            'lede': lede,
            'sections': sections,
            'quote': quote
        }

with open('/Users/charlotte/Desktop/NGÂN HÀ/CTY TechGalaxy Group/NganHa-WebBooking/src/data/DiscoveryData.json', 'w') as out:
    json.dump(data, out, indent=2, ensure_ascii=False)

print("Parsed and saved to DiscoveryData.json")
